spyder/workers/updates.py

```diff
@@ -225,16 +227,38 @@ def _progress_reporter(self, block_number, read_size, total_size):
     def _download_installer(self):
         """Donwload latest Spyder standalone installer executable."""
         tmpdir = tempfile.gettempdir()
-        is_full_installer = (is_module_installed('numpy') or
-                             is_module_installed('pandas'))
-        if os.name == 'nt':
-            name = 'Spyder_64bit_{}.exe'.format('full' if is_full_installer
-                                                else 'lite')
+
+        # If running on macOS under Rosetta, platform.machine will not be
+        # correct so check for ARM64 in version instead. All other
+        # platforms will be x86_64.
+        mach = "arm64" if "ARM64" in platform.version() else "x86_64"
+
+        if parse(self.latest_release_version) > parse("5"):
+            # conda-based installer name
+            if os.name == 'nt':
+                plat, ext = 'Windows', 'exe'
+            elif platform.system == 'Darwin':
+                plat, ext = 'macOS', 'pkg'
+            else:
+                plat, ext = 'Linux', 'sh'
+            name = f'Spyder-{plat}-{mach}.{ext}'
         else:
-            name = 'Spyder{}.dmg'.format('' if is_full_installer else '-Lite')
+            # 5.x installer name
+            is_full_installer = (is_module_installed('numpy') or
+                                 is_module_installed('pandas'))
+            if os.name == 'nt':
+                name = 'Spyder_64bit_{}.exe'.format(
+                    'full' if is_full_installer else 'lite'
+                )
+            else:
+                name = 'Spyder{}{}.dmg'.format(
+                    '_{mach}' if mach == 'arm64' else '',
+                    '' if is_full_installer else '-Lite'
```

Comment:
The other branch has explicit checks for Windows and macOS and a fallback to the *nix based installers, while this one only has an explicit check for Windows and falls back to macOS—and I don't see any reason at least internal to this function why the second branch would only get executed on Windows and macOS while the first branch would not. Seems like this should be consistent here between branches, either to avoid logic/correctness issues or at least for clarity/consistency's sake?

Reply:
Okay, I will take a look.

Reply:
The `_download_installer` method is only called on Linux platforms when the new version is >= 6.0. So when the version is >=6.0, then the artifact name is determined for all three platforms. When the new version is less than 6.0, `_download_installer` is only called for Windows and macOS platforms, to the second branch will only be executed on Windows and macOS platforms.

Reply:
Okay, I guess so, as long as we're 100% sure it will never get called in that case